ocs_ci/ocs/platform_nodes.py

```diff
@@ -1272,3 +1308,137 @@ def get_worker_resource_id(self, resource):
+        self.compute_count = compute_count
+        self.current_compute_count = len(get_typed_nodes())
+        self.target_compute_count = (self.current_compute_count
+                                     + self.compute_count)
+        self.previous_dir = os.getcwd()
+        self.terraform_data_dir = os.path.join(
+            self.cluster_path,
+            constants.TERRAFORM_DATA_DIR
+        )
+        self.terraform_work_dir = constants.VSPHERE_DIR
+        self.terraform = Terraform(self.terraform_work_dir)
+        self.upi_repo_path = os.path.join(
+            constants.EXTERNAL_DIR, 'installer',
+        )
+
+    def _update_terraform(self):
+        """
+        Update terraform variables
+        """
+        logger.debug("Updating terraform variables")
+        self.terraform_var = os.path.join(
+            self.cluster_path,
+            constants.TERRAFORM_DATA_DIR,
+            "terraform.tfvars"
+        )
+        compute_str = 'compute_count ='
+        to_change = f"{compute_str} \"{self.current_compute_count}\""
+        updated_compute_str = f"{compute_str} \"{self.target_compute_count}\""
+        logging.debug(f"Updating {updated_compute_str} in {self.terraform_var}")
+
+        # backup the terraform variable file
+        original_file = f"{self.terraform_var}_{int(time.time())}"
+        shutil.copyfile(self.terraform_var, original_file)
+        logging.info(f"original terraform file: {original_file}")
+
+        replace_content_in_file(
+            self.terraform_var,
+            to_change,
+            updated_compute_str
+        )
+
+    def _update_machine_conf(self):
+        """
+        Update the machine config for vsphere
+        """
+        to_change = "clone {"
+        add_file_block = f"{constants.LIFECYCLE}\n  {to_change}"
+        logging.debug(
+            f"Adding {constants.LIFECYCLE} to"
+            f" {constants.INSTALLER_MACHINE_CONF}"
+        )
+        replace_content_in_file(
+            constants.INSTALLER_MACHINE_CONF,
+            to_change,
+            add_file_block
+        )
+
+        # change root disk size
+        change_vm_root_disk_size(constants.INSTALLER_MACHINE_CONF)
+
+        # update gateway and DNS
+        if config.ENV_DATA.get('gateway'):
+            replace_content_in_file(
+                constants.INSTALLER_IGNITION,
+                '${cidrhost(var.machine_cidr,1)}',
+                f"{config.ENV_DATA.get('gateway')}"
+            )
+
+        if config.ENV_DATA.get('dns'):
+            replace_content_in_file(
+                constants.INSTALLER_IGNITION,
+                constants.INSTALLER_DEFAULT_DNS,
+                f"{config.ENV_DATA.get('dns')}"
+            )
+
+        # update the zone in route
+        if config.ENV_DATA.get('region'):
+            def_zone = 'provider "aws" { region = "%s" } \n' % config.ENV_DATA.get('region')
+            replace_content_in_file(constants.INSTALLER_ROUTE53, "xyz", def_zone)
+
+    def add_node(self):
```

Comment:
Does creation of vm instances and attach happens at the same time ? if not how about splitting into functions like create phase and attach phase?

Also, if this can be split , just to be consistent, can we rename this add_node. In AWS I named those functions as prepare_node() and attach_nodes(). Not sure if its relevant to vmware platform but if possible just to be consistent in naming conventions we can have same method names so that it improves readability. I am also ok to change name in AWS classes if you feel this 'add_node' is apt.

Reply:
there i sno separation for creation and attach in vsphere, both happens at same. So this PR doesn't have separate function for create and attach. Even though it doesn't have separate function, I use same approach as AWS implementaion, so that end user doesn't need to chnage anything w.r.t to vsphere.

How to add node:
```
plt = PlatformNodesFactory()
node_util = plt.get_nodes_platform()
node_util.create_and_attach_nodes_to_cluster({}, "RHCOS", num_nodes)
```